github_issue_adapter/query-visitors.py

```diff
@@ -0,0 +1,137 @@
+        if tokens:
+            token = tokens[0]
+            Console.stderr(f'Using GitHub token from environment variable "{token}".');
+            return os.environ[token]
+
+        key = 'github_oauth_access_token'
+        with open(self.credentials_file) as f:
+            def extract_setting(line):
+                k, v = line.strip().split('=')
+                return k.strip(), v.strip()
+            settings = (extract_setting(line) for line in f)
+            settings = { k: v for k, v in settings }
+
+        if key not in settings:
+            from inspect import cleandoc
+            Console.stderr(cleandoc(
+                """
+                Could not find token for access to GitHub.
+                Please either create file {file} with entry {key}=<GitHub token>
+                or set one of the following environment variables: {vars}.
+                """.format( file=self.credentials_file, key=key,
+                            vars=', '.join(self.token_vars))))
+            sys.exit(1)
+
+        if not self.is_quiet:
+            Console.stderr(f'Using GitHub token from entry {key}' \
+                           + f' in file {self.credentials_file}.');
+        return settings[key]
+
+
+class MergeStatement:
+
+    def __init__(self, db_schema, db_table):
+        self.db_schema = db_schema
+        self.db_table = db_table
+        self.rows = []
+        self.is_verbose = False
+
+    def verbose(self, is_verbose: bool):
+        self.is_verbose = is_verbose
+
+    def add_row(self, repo, date, count, uniques):
+        row = f"('{repo}', '{date}', '{count}', '{uniques}')"
+        self.rows.append(row)
+        if self.is_verbose:
+            print(f'  - {date}: {count}, {uniques}')
+
+    def build(self) -> str:
+        return f'merge into {self.db_schema}.{self.db_table} T\n' \
+            + 'using values\n' \
+            + '     ' + ',\n     '.join(self.rows) + '\n' \
+            + 'U ("REPO", "DATE", "COUNT", "UNIQUES")\n' \
+            + 'on (T."REPO" = U."REPO" AND T."DATE" = U."DATE")\n' \
+            + 'when matched then update set\n' \
+            + '     T."COUNT" = U."COUNT",\n' \
+            + '     T."UNIQUES" = U."UNIQUES",\n' \
+            + '     T."UPDATED" = CURRENT_TIMESTAMP\n' \
+            + 'when not matched then insert values (\n' \
+            + '      U."REPO", U."DATE", U."COUNT", U."UNIQUES",\n' \
+            + '      CURRENT_TIMESTAMP, CURRENT_TIMESTAMP\n' \
+            + ');'
+
+def main():
+    args = create_parser().parse_args()
+    token_reader = GithubTokenReader()
+    token_reader.quiet(args.quiet)
+    token = token_reader.get_token()
+    exasol = database_connection(db_host, db_user, args.password)
+    fetcher = GithubIssuesFetcher('exasol', token)
+    if args.repositories:
+        repos = args.repositories
+    else:
+        repos = fetcher.list_repositories()
+    i = len(repos)
+    for repo in repos:
```

Comment:
```suggestion
    for i, repo in enumerate(repos):
```
if you need the remaining `remaining = i - len(repos)`